grudge/trace_pair.py

```diff
@@ -275,58 +280,49 @@ def connected_ranks(dcoll: DiscretizationCollection):
 class _RankBoundaryCommunication:
     base_tag = 1273
 
-    def __init__(self, dcoll: DiscretizationCollection,
-                 remote_rank, vol_field, tag=None):
-        self.tag = self.base_tag
-        if tag is not None:
-            self.tag += tag
-
-        self.dcoll = dcoll
-        self.array_context = vol_field.array_context
-        self.remote_btag = BTAG_PARTITION(remote_rank)
-        self.bdry_discr = dcoll.discr_from_dd(self.remote_btag)
+    def __init__(self,
+                 dcoll: DiscretizationCollection,
+                 array_container: ArrayOrContainerT,
+                 remote_rank, tag=None):
+        actx = get_container_context_recursively(array_container)
+        btag = BTAG_PARTITION(remote_rank)
 
-        from grudge.op import project
+        local_array_ctr = project(dcoll, "vol", btag, array_container)
```

Comment:
```suggestion
        local_bdry_data = project(dcoll, "vol", btag, array_container)
```
Or some such. (also rename the attribute)

Reply:
Done: https://github.com/inducer/grudge/pull/154/commits/7b0729a5e17eacae466a46e285513cb5fa62e446